doc/designs/application-specific-passwords.md

```diff
@@ -0,0 +1,283 @@
+# Application-specific passwords
+
+**DESIGN STAGE**
+
+## Overview
+
+FreeIPA does not currently allow users to have more than one password.
+Using a single password for the user's account carries potential risk and
+inconvenience with it. If the password is compromised on only one of the devices
+the user uses, the attacker gains full administrative access to the user's
+account and all services associated with it. The user can then halt it only by
+changing this single account's password and reconfiguring all systems and
+applications using that password for authentication (even if only to one
+particular service) to use the new one.
+
+This enhancement therefore brings support multiple passwords for a single user
+and extends the HBAC access control to allow specifying what application-specific
```

Comment:
Would it make sense to bind application-specific passwords to HBAC services and service groups?
- The application name must be a service name or a service group (``cn=hbacservicegroups,cn=hbac,$SUFFIX`` or ``cn=hbacservices,cn=hbac,$SUFFIX``)
- Application-specific passwords only allow access to a matching service category.

Reply:
Maybe. This would mean that the user would be able to create app-specific passwords only for the services or service groups that already have a record in the locations you pointed out. The questions are, whether this is desirable, and what would this do with the complexity of the implementation.
An advantage of this approach is, obviously, that on top of this, there would be no need to do anything else with HBAC - neither during implementation (this would render the proposal to extend HBAC with an ability to specify allowed LDAP subtrees for app-specific passwords redundant), nor on the part of administrators, as existing HBAC rules would automatically apply for app-specific passwords.